packages/flutter_tools/lib/src/ios/core_devices.dart

```diff
@@ -0,0 +1,847 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/logger.dart';
+import '../base/process.dart';
+import '../convert.dart';
+import '../device.dart';
+import '../macos/xcode.dart';
+
+/// A wrapper around the `devicectl` command line tool.
+///
+/// CoreDevice is a device connectivity stack introduced in Xcode 15. Devices
+/// with iOS 17 or greater are CoreDevices.
```

Comment:
maybe also document what was used before ios 17?

Reply:
I added a comment for the `_startAppOnCoreDevice` function in ios/devices.dart. I thought it made more sense to add it there, since there's not exactly a one-to-one previous equivalent for CoreDevices